src/NLog/Layouts/CsvLayout.cs

```diff
@@ -115,12 +115,13 @@ public CsvLayout()
         /// </summary>
         protected override void InitializeLayout()
         {
-            base.InitializeLayout();
             if (!WithHeader)
             {
                 Header = null;
             }
 
+            base.InitializeLayout();
```

Comment:
what's the reason for this moved line?

Reply:
InitializeLayout performs scanning of all referenced objects. No need to scan the Header-property when WithHeader = false.

Reply:
👍